api/envoy/service/discovery/v2/tds.proto

```diff
@@ -33,7 +39,7 @@ service RuntimeDiscoveryService {
 }
 
 // TDS resource type. This describes a layer in the runtime virtual filesystem.
-// [#not-implemented-hide:]
 message Runtime {
-  google.protobuf.Struct layer = 1;
+  string name = 1;
```

Comment:
Should this be required for admin printing, etc.? Should we also enforce it is unique when sent down?

Reply:
Sure, can add these.

Reply:
Actually, I take that back; there is already a `name` for admin printing in https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/bootstrap/v2/bootstrap.proto#L276. This is where the non-emptiness should be enforced, I can add this. The `name` in `tds.proto` is the resource self-description, e.g. similar to what we have in `Cluster`. This is unique by definition.